tls/s2n_record_read.c

```diff
@@ -95,6 +95,13 @@ int s2n_record_header_parse(
     POSIX_GUARD(s2n_stuffer_read_bytes(in, protocol_version, S2N_TLS_PROTOCOL_VERSION_LEN));
 
     const uint8_t version = (protocol_version[0] * 10) + protocol_version[1];
+    /* We record the protocol version in the first record seen by the server for fingerprinting usecases.
+     * actual_protocol_version_established is set after writing the server hello.
+     */
+    if (!conn->actual_protocol_version_established) {
+        conn->client_hello_record_version = version;
+    }
```

Comment:
Good question. I suppose with my current implementation, record 2 would overwrite record 1's version. But I'm not sure that's an issue; do you think there is a correct version to grab in this scenario?

Reply:
If an implementation changed this version between the two records that they sent, I imagine users might expect the first record version to be returned, since this is the initial client hello record. Not sure, though.

If this is the case, we could maybe track it with a single bit on the client hello maybe?
https://github.com/aws/s2n-tls/blob/c74f442b55589872b1374559bb03878c49761ca6/tls/s2n_client_hello.h#L35-L37

Reply:
I feel like it might make more sense to return the first version in this case. If an implementation were to send two client hello records, I could see them maybe changing the first record version for compatibility but not the second, and in this case I think users might expect the first version to be returned. Not sure, though.